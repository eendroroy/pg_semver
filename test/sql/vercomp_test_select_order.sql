LOAD 'vercomp';

CREATE TABLE versions(version VERSION);

INSERT INTO versions VALUES ('1.0.0'), ('0.0.0'), ('2.5.0-beta1'), ('2.0.0-rc1'),('2.10.0-beta0'), ('20.2.0-alpha'),
                            ('30.0.0'),  ('3.0.0'),  ('3.0.0-rc.2'),  ('3.0.0-rc.0'),  ('3.0.0-b.2'),  ('3.0.0-a.0');


SELECT * FROM versions;
SELECT * FROM versions ORDER BY version;
SELECT * FROM versions ORDER BY version DESC;

DROP TABLE IF EXISTS versions;
