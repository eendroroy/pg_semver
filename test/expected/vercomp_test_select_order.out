LOAD 'vercomp';
CREATE TABLE versions(version VERSION);
INSERT INTO versions VALUES ('1.0.0'), ('0.0.0'), ('2.5.0-beta1'), ('2.0.0-rc1'),('2.10.0-beta0'), ('20.2.0-alpha'),
                            ('30.0.0'),  ('3.0.0'),  ('3.0.0-rc2'),  ('3.0.0-rc0'),  ('3.0.0-beta2'),  ('3.0.0-alpha0');
SELECT * FROM versions;
   version    
--------------
 1.0.0
 0.0.0
 2.5.0-beta1
 2.0.0-rc1
 2.10.0-beta0
 20.2.0-alpha
 30.0.0
 3.0.0
 3.0.0-rc2
 3.0.0-rc0
 3.0.0-beta2
 3.0.0-alpha0
(12 rows)

SELECT * FROM versions ORDER BY version;
   version    
--------------
 0.0.0
 1.0.0
 2.0.0-rc1
 2.5.0-beta1
 2.10.0-beta0
 3.0.0-alpha0
 3.0.0-beta2
 3.0.0-rc0
 3.0.0-rc2
 3.0.0
 20.2.0-alpha
 30.0.0
(12 rows)

SELECT * FROM versions ORDER BY version DESC;
   version    
--------------
 30.0.0
 20.2.0-alpha
 3.0.0
 3.0.0-rc2
 3.0.0-rc0
 3.0.0-beta2
 3.0.0-alpha0
 2.10.0-beta0
 2.5.0-beta1
 2.0.0-rc1
 1.0.0
 0.0.0
(12 rows)

DROP TABLE IF EXISTS versions;
